src/Razor/src/Microsoft.AspNetCore.Razor.Language/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -1348,6 +1348,22 @@ public override void VisitCSharpTemplateBlock(CSharpTemplateBlockSyntax node)
             // We need to capture this in the IR so that we can give each piece the correct source mappings
             public override void VisitCSharpExplicitExpression(CSharpExplicitExpressionSyntax node)
             {
+                if (_builder.Current is HtmlAttributeIntermediateNode)
+                {
+                    // This can happen inside a data- attribute
+                    _builder.Push(new CSharpExpressionAttributeValueIntermediateNode()
```

Comment:
Will this result in `data-` attributes being treated as conditional attributes? Aka `<input data-checked="@false" />` => `<input />`. Just double checking that we're not breaking a pre-existing language feature.

Reply:
Yes, and we want that for components. This code path is for components only.

I think we need to review this broadly.